coresdk/src/test/unit_tests/unit_test_json.cpp

```diff
@@ -117,7 +117,18 @@ TEST_CASE("json can be created and read", "[json]")
 
         REQUIRE("#00ff00ff" == color_to_string(deserialized_clr));
     }
+    SECTION("json can be created and read with different number types")
+    {
+        json number_types = create_json();
+
+        json_set_number(number_types, "float", static_cast<float>(21.2));
+        json_set_number(number_types, "double", static_cast<double>(30.1));
 
+        REQUIRE(json_read_number(number_types, "float") == static_cast<float>(21.2));
+        REQUIRE(json_read_number_as_double(number_types, "double") == static_cast<double>(30.1));
```

Comment:
For floating point comparisons it'd be good if we use matchers to avoid issues with different floating point representations of a number not being exactly equal. There are a few different kinds ([see Catch2 docs for more info](https://github.com/catchorg/Catch2/blob/v2.x/docs/matchers.md)), but I've been using WithinRelMatcher with the default epsilon, which is covenient.
You can use this by adding a using declaration for the relevant matcher and then using the `REQUIRE_THAT(value/object, matcher expression)` syntax. For example, for the WithinRelMatcher:
```suggestion
using Catch::Matchers::WithinRel;
...
        REQUIRE_THAT(json_read_number(number_types, "float"), WithinRel(21.2f));
        REQUIRE_THAT(json_read_number_as_double(number_types, "double"), WithinRel(30.1));
```
Note the floating-point literal suffix f here, too.